### DISCUSSION TITLE
更明确的兼容环境定义？

### DISCUSSION THREAD

**Discussion by dislido** [ant-design/ant-design#51701] [Ideas]
根据https://github.com/ant-design/ant-design?tab=readme-ov-file#-environment-support ，antd在所有环境的兼容性都是last 2 versions，但实际上antd会考虑比这更早的浏览器的兼容性，建议更明确的声明antd的兼容环境版本：
- antd直接支持的各环境版本号，可以不是当前实际支持的最低版本号，而是目标兼容版本 如以2024年初做基线-chrome/edge121,safari17.3,firefox122，即使当前版本实际支持更低浏览器版本，也不需要保证之后版本会兼容
- 可通过polyfill等方式兼容到的最低版本（或许不是很容易定义？）

这个兼容版本可以在antd的major版本进行提升

这么做可以带来的好处：
- 贡献者可以明确知道能够使用哪些特性，如使用某些新特性可以带来性能提升或更简洁的代码，但有一定兼容性要求，不明确能不能使用的情况下开发者可能会倾向于用旧特性实现
- 可以放心移除不再需要的兼容代码，或将旧逻辑重构为用新特性实现，带来打包体积/性能/仓库体积的优化
- 使用者可以明确自己支持的浏览器版本，比如移动端app webview等不那么容易在用户设备上更新的环境

**Comment by petercat-assistant** [comment]
您提出的建议关于更明确地声明 Ant Design 的兼容环境版本是非常有意义的。明确的兼容性声明可以帮助开发者和用户更好地理解和使用库的功能。以下是您建议的几点总结：

1. **明确支持的环境版本**：建议声明 Ant Design 直接支持的各环境版本号，作为目标兼容版本。即使当前版本实际支持更低的浏览器版本，也不需要保证之后版本会兼容。

2. **通过 Polyfill 兼容的最低版本**：虽然定义可能不容易，但可以考虑声明通过 Polyfill 等方式兼容到的最低版本。

3. **版本提升策略**：在 Ant Design 的 major 版本中提升兼容版本。

这样做的好处包括：
- 贡献者可以明确知道能够使用哪些新特性，从而优化性能和代码简洁性。
- 可以移除不再需要的兼容代码，优化打包体积和性能。
- 使用者可以明确自己支持的浏览器版本，尤其是在更新不便的环境中。

如果您希望将此建议提交给 Ant Design 团队，您可以在 GitHub 上创建一个讨论或 issue，以便开发团队考虑和讨论这一建议。

如需进一步帮助，请在评论中描述您的问题并 @petercat-assistant 以开始与我对话。

**Comment by afc163** [comment]
可以看 https://ant.design/docs/react/compatible-style-cn

**Comment by zombieJ** [comment]
参考：https://github.com/ant-design/ant-design/issues/50265#issuecomment-2270259147

> 最近两个版本的意思就是当你遇到 BUG 时，我们会对在最近版本出现的问题进行维护。而对于过于古老的浏览器版本则不保证提供维护，从而保证社区成员有足够的精力。所以 antd 支持的版本可以理解为是超过约定版本的，但是并不会承诺具体维护到哪个古老的版本为止。

**Comment by dislido** [comment]
了解，感谢大佬们解答
